frame/contracts/src/wasm/runtime.rs

```diff
@@ -1254,6 +1260,35 @@ define_env!(Env, <E: Ext>,
 		)?)
 	},
 
+	// Checks whether a specified address belongs to a contract.
+	//
+	// # Parameters
+	//
+	// - account_ptr: a pointer to the address of the beneficiary account
+	//   Should be decodable as an `T::AccountId`. Traps otherwise.
+	//
+	// Returned value is a u32-encoded boolean: (0 = false, 1 = true).
+	[__unstable__] seal_is_contract(ctx, account_ptr: u32) -> u32 => {
+		ctx.charge_gas(RuntimeCosts::IsContract)?;
+		let address: <<E as Ext>::T as frame_system::Config>::AccountId =
+			ctx.read_sandbox_memory_as(account_ptr)?;
+
+		Ok(ctx.ext.is_contract(address) as u32)
+	},
+
+	// Checks whether the caller of the current contract is the origin of the whole call stack.
+	//
+	// Prefer this over `seal_is_contract` when checking whether being called by a contract
+	// or a plain account. Reason is that it performs better since it does not need to
+	// do any storage lookups: `true` implies that this contract is being called by a plain account
+	// and `false` implies that the caller is another contract.
```

Comment:
```suggestion
	//
	// Prefer this over `seal_is_contract` when checking whether your contract is being called by a contract
	// or a plain account. The reason is that it performs better since it does not need to
	// do any storage lookups.
	//
	// A return value of`true` indicates that this contract is being called by a plain account
	// and `false` indicates that the caller is another contract.
```